web/packages/teleterm/src/services/pty/ptyService.ts

```diff
@@ -23,21 +23,26 @@ import { RuntimeSettings } from 'teleterm/mainProcess/types';
 import { buildPtyOptions } from './ptyHost/buildPtyOptions';
 import { createPtyHostClient } from './ptyHost/ptyHostClient';
 import { createPtyProcess } from './ptyHost/ptyProcess';
-import { PtyServiceClient, SshOptions } from './types';
+import { PtyServiceClient, PtyOptions } from './types';
+import { getWindowsPty } from './ptyHost/windowsPty';
 
 export function createPtyService(
   address: string,
   credentials: ChannelCredentials,
   runtimeSettings: RuntimeSettings,
-  sshOptions: SshOptions
+  options: PtyOptions
 ): PtyServiceClient {
   const ptyHostClient = createPtyHostClient(address, credentials);
 
   return {
     createPtyProcess: async command => {
+      const windowsPty = getWindowsPty(runtimeSettings, options.terminal);
       const { processOptions, creationStatus } = await buildPtyOptions(
         runtimeSettings,
-        sshOptions,
+        {
+          ssh: options.ssh,
+          terminal: { useConpty: windowsPty.backend === 'conpty' },
```

Comment:
Why return a string from `getWindowsPty` but then use a boolean to pass that information through RPC? Wouldn't it be better to either pass a string or a boolean all the way through?

Reply:
Ideally, I'd pass a union of `conpty` | `winpty` all the way through, but unfortunately it would get lost when going through RPC (I could use enums, but they are less convenient to use and I'd still have to convert them to strings for xterm). Because of that, I decided to send a boolean through RPC. 
I see the confusion however, so now I send a boolean everywhere. I also had to adjust the types a bit, previously I passed `TerminalOptions` to `buildPtyOptions` but now we need a more specific type there.